api/ccapi/option3_hide_updated.h

```diff
@@ -0,0 +1,83 @@
+// Layer.h (ccapi)
+class Layer {
+  public:
+    Layer();
+    virtual foo() = 0;
+    virtual bar() = 0;
+};
+
+// LayerInternal.h (internal)
+class LayerInternal : public Layer {
+  public:
+    LayerInternal();
+    virtual foo() { /** do some work */ }
+
+  private:
+    // Actual data members
+    // weights, etc
+};
+
+// layer_factory.h (ccapi)
+std::unique_ptr<Layer> createLayer(string layer_type, std::vector<string> properties);
+
```

Comment:
Do we need to vector<string> for property? Args may be?

Reply:
Args will be templated, which will need including nntrainer internal headers in API header.

Reply:
PropertyHandler might come to a viable alternative to std::vector<string>